### PR TITLE
Add support for specifying a namespace for the resource to be created in

### LINKED ISSUE #55: Need ability to specify namespace in CR
## What happens?

Currently all resources get created within the namespace the job runs in. This makes it impossible to e.g. create a new namespace, create a new SA and CR within it, then have the operator take it from there.

## What were you expecting to happen?

We can create resources in any namespace, as long as the SA has sufficient access.
We should extend the CRD and add:
```
spec:
  namespace: "xxx"
```
That defaults to the current namespace if it's not defined.
The job templates, and `discoverEnvironment.sh` and `resourceManager.sh` need to set the proper context.

This can simply be passed as an ENV variable to the jobs. The variable already exists, but is not consumed.
```
{{ if .Config.Spec.namespace }}
              value: {{ .Config.Spec.namespace }}
{{ else }}
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace          
{{ end }}
```

## Steps to reproduce: 

Try creating a resource in another namespace.

### PR DESCRIPTION
**Pull request by Smiley73**
## Description

Creation of resources in other namespaces was not possible. This PR adds a new optional property `namespace` to be able to specify a specific target namespace.

Fixes #55

## Type of change

Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: deploy/kubernetes/operator.yaml [hunk 1]
```diff
@@ -19,7 +19,8 @@ spec:
           image: quay.io/kohlstechnology/eunomia-operator:v0.0.1
           command:
           - eunomia-operator
-          imagePullPolicy: Always
+          #imagePullPolicy: Always
```
Comment (seanmalloy) [inline deploy/kubernetes/operator.yaml]: Please delete this commented line.

File: templates/cronjob.yaml [hunk 2]
```diff
@@ -12,12 +12,17 @@ spec:
           containers:
           - name: template-processor
             imagePullPolicy: Always
+            #imagePullPolicy: IfNotPresent
```
Comment (seanmalloy) [inline templates/cronjob.yaml]: Please remove commented code.

File: templates/job.yaml [hunk 3]
```diff
@@ -11,14 +11,19 @@ spec:
       containers:
       - name: template-processor
         imagePullPolicy: Always
+        #imagePullPolicy: IfNotPresent
```
Comment (seanmalloy) [inline templates/job.yaml]: Same here. Please remove this.

**Comment by seanmalloy** [comment]
Based on comment https://github.com/KohlsTechnology/eunomia/issues/55#issuecomment-505649886 from @sanbornick we will need to discuss this further before merging.

**Comment by Smiley73** [comment]
I think I had must have had something misconfigured last time I tried this. I got it working now with the existing functionality. Let me close this issue.

**Pull request closed without merging**